### PR TITLE
Feature/add in range indicator and other features

### PR DESCRIPTION
**Pull request by nicholas-yong**
## Overview
Adds the following features to the package:
- Some dependency updates.
- Removed unused commented code.
- Removed autoRange prop.
- Modified the range selector to use only one window instead of two.
- Modified the range selector to have a highlighted selector when hovering over buttons.
- Modified the onChange function to return the new state of the control.
- Added some utility functions

## Video View
https://www.loom.com/share/9d04e0fde3a9434fb07697aabf8d351d?sid=4d2f1bcd-abe4-4283-878b-4d5de4ee16a4

### CODE REVIEW

File: lib/month-picker.js [hunk 1]
```diff
@@ -17,65 +18,47 @@
...
 });
 exports["default"] = void 0;
-
 var _react = _interopRequireWildcard(require("react"));
-
 var _propTypes = _interopRequireDefault(require("prop-types"));
-
 var _reactTapper = _interopRequireDefault(require("react-tapper"));
-
+var _debug = _interopRequireDefault(require("debug"));
```
Comment (venkat-balachandran) [inline lib/month-picker.js:28]: suggestion: you need not check in lib file changes, u already made changes in src folder

File: src/month-picker.jsx [hunk 2]
```diff
@@ -309,43 +264,166 @@ export default class MonthPicker extends Component {
...
+     * @param {boolean} isSameYear 
+     * Whether the displayed year is the same year as the selected year
+     */
+    setHoverState(dataId, positiveIteration, selectedMonth, isSameYear)
+    {
+        if(positiveIteration)
+        {
+            for(let i = 1; i <= 12; i += 1)
+            {
+                const element = document.querySelector(`[data-id="0:${i}"]`);
```
Comment (venkat-balachandran) [inline src/month-picker.jsx]: suggestion: would it be possible to use a ref here? or is there a limitation to use dom element directly?

File: src/month-picker.jsx [hunk 3]
```diff
@@ -309,43 +264,166 @@ export default class MonthPicker extends Component {
...
+                if(element) 
+                {
+                    switch(true)
+                    {
+                        case (i < dataId):
+                            element.classList.remove(_SELECT_CLASS_NAME);
```
Comment (venkat-balachandran) [inline src/month-picker.jsx]: suggestion: would it be possible to do this with classNames?

**Review by venkat-balachandran — COMMENTED** [review]
Just had some general questions around handling DOM elements directly

File: lib/month-picker.js (hunk 1 above)
Comment (nicholas-yong) [reply lib/month-picker.js:28]: I wasn't entirely sure if I needed to as Webapp is currently reading the package from the Github url 😓

File: src/month-picker.jsx (hunk 3 above)
Comment (nicholas-yong) [reply src/month-picker.jsx]: ah, do you mean the classnames library? I can take a look. It's only the 'select' classname being modified though.

File: src/month-picker.jsx (hunk 2 above)
Comment (nicholas-yong) [reply src/month-picker.jsx]: I can take a look to see how difficult it'd be using a ref, I just thought it was cleaner to use a local variable as it's not referenced anywhere else.

File: src/month-picker.jsx (hunk 3 above)
Comment (venkat-balachandran) [reply src/month-picker.jsx]: upto you :) , this is a third party library so i am not fussed

File: src/month-picker.jsx (hunk 2 above)
Comment (venkat-balachandran) [reply src/month-picker.jsx]: if it takes a while leave it

File: Gruntfile.js [hunk 4]
```diff
@@ -1,113 +1,111 @@
 "use strict";
 
-const sass = require('node-sass')
+const sass = require("sass");
```
Comment (nicholas-yong) [inline Gruntfile.js:3]: Outside of this change, the other changes are a result of running prettier.

File: webpack.config.js [hunk 5]
```diff
@@ -1,47 +1,47 @@
 "use strict";
 
-const path = require('path')
+const path = require("path");
```
Comment (nicholas-yong) [inline webpack.config.js:3]: prettier was run here.

File: lib/month-picker.js (hunk 1 above)
Comment (nicholas-yong) [reply lib/month-picker.js:28]: Yeah, it turns out it is needed as it is reading from the Github URL, the other files aren't needed though.

**Review by lizacodes — APPROVED** [review]

**Pull request merged**